src/Symfony/Bundle/FrameworkBundle/Command/CacheClearCommand.php

```diff
@@ -26,6 +29,42 @@
  */
 class CacheClearCommand extends ContainerAwareCommand
 {
+    private $filesystem;
+    private $cacheClearer;
+    private $cacheDir;
+
+    /**
+     * @param Filesystem            $filesystem
+     * @param CacheClearerInterface $cacheClearer
+     * @param string                $cacheDir
+     */
+    public function __construct($filesystem = null, CacheClearerInterface $cacheClearer = null, $cacheDir = null)
+    {
+        parent::__construct();
+
+        if (!$filesystem instanceof Filesystem) {
```

Comment:
For assets:installs that makes $baseDir the 1st argument, which is as $name also string. Ie we cant detect if user passes name (old api) or path (new api).

I like $filesystem being last and optional though. Suggestions?